### PR TITLE
MAB ordering

### PR DESCRIPTION
**Pull request by C-gyorfi**
### Context

image

### CODE REVIEW

File: app/controllers/mac_authentication_bypasses_controller.rb [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-                                     @search = params[:search]
-                                     MacAuthenticationBypass.where(
+                                     results = MacAuthenticationBypass.where(
                                        "name LIKE ? or description LIKE ? or address LIKE ?",
                                        "%#{@search}%", "%#{@search}%", "%#{@search}%"
                                      ).page(params[:page])
+
+                                     params[:sort] ? results.order("#{@sort[:sort_by]} #{@sort[:sort_order]}") : results
+                                   elsif params[:sort]
+                                     MacAuthenticationBypass.order("#{@sort[:sort_by]} #{@sort[:sort_order]}").page params[:page]
```
Comment (emileswarts) [inline app/controllers/mac_authentication_bypasses_controller.rb:15]: Can this be sanitized with "?" syntax?

Comment (C-gyorfi) [reply app/controllers/mac_authentication_bypasses_controller.rb]: `.order` doesn't seem to support that syntax, I hand-rolled a solution, which may be overkill, let me know what you think cc3a83d

File: app/controllers/mac_authentication_bypasses_controller.rb [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
                                        "name LIKE ? or description LIKE ? or address LIKE ?",
                                        "%#{@search}%", "%#{@search}%", "%#{@search}%"
                                      ).page(params[:page])
+
+                                     params[:sort] ? results.order(order_by(@sort)) : results
+                                   elsif params[:sort]
+                                     MacAuthenticationBypass.order(order_by(@sort)).page params[:page]
                                    else
                                      MacAuthenticationBypass.page params[:page]
                                    end
```
Comment (emileswarts) [inline app/controllers/mac_authentication_bypasses_controller.rb:18]: Would something like this be equivalent to the above?

```ruby
search = params.fetch(:search, "")
sort = params.fetch(:sort, "created_at")
sort_order = params.fetch(:sort_order, "desc")

@mac_authentication_bypasses = MacAuthenticationBypass.where(
                                   "name LIKE ? or description LIKE ? or address LIKE ?",
                                   "%#{@search}%", "%#{@search}%", "%#{@search}%"
                                 ).order_by("#{sort} #{sort_order}").page(params[:page])
```

Comment (emileswarts) [reply app/controllers/mac_authentication_bypasses_controller.rb:18]: Also wondering if we can push some of this stuff down into the model.
Maybe with a search class method like: 

MacAuthenticationBypass.search(term: term, sort: sort, sort_order: sort_order)

File: app/controllers/mac_authentication_bypasses_controller.rb [hunk 3]
```diff
@@ -99,4 +103,31 @@ def publish_authorised_macs
...
+
+  def sortable_attributes
+    {
+      address: "address",
+      name: "name",
+      description: "description",
+      created_at: "created_at",
+      updated_at: "updated_at",
+    }
+  end
```
Comment (emileswarts) [inline app/controllers/mac_authentication_bypasses_controller.rb:128]: I'd be tempted to remove this mapping if it's exactly the same, just symbols vs strings.

**Pull request closed without merging**